src/rgw/rgw_sal_motr.cc

```diff
@@ -1465,7 +1465,11 @@ int MotrObject::set_obj_attrs(const DoutPrefixProvider* dpp, RGWObjectCtx* rctx,
 
 int MotrObject::get_obj_attrs(RGWObjectCtx* rctx, optional_yield y, const DoutPrefixProvider* dpp, rgw_obj* target_obj)
 {
-  if (this->category == RGWObjCategory::MultiMeta)
+  req_state *s = (req_state *) rctx->get_private();
+  /* handle get object for multipart */
+  string req_method = s->info.method;
+  ldpp_dout(dpp, 20) << "req_method " << req_method << dendl;
+  if (this->category == RGWObjCategory::MultiMeta && req_method != "GET")
```

Comment:
need to avoid this for anything (PUT/POST/HEAD/DELETE etc) request type? We decided to avoid only for PUT or if needed for POST also. Dont extend the impact unless absolutely necessary !

Reply:
@suryakumar1024 : Test this change (@line 1472) thoroughly.

Reply:
Hi @cdeshmukh I wanted to Enable the get_obj_attrs for only for GET request type and that to only for MultiMeta type objects. 

**if (this->category == RGWObjCategory::MultiMeta && req_method != "GET")**

This above code change works fine. And it won't affect the existing code flow, instead it just enable only if the request type was GET and the object type is MultiMeta.

Reply:
> @suryakumar1024 : Test this change (@line 1472) thoroughly.

HI @DPG17 Sure as discussed over chat. I have tested this code change with all tagging/ attrs related scenarios, Will post those results in PR comments.